drivers/overlay/ov_network.go

```diff
@@ -46,10 +47,15 @@ type network struct {
 	once      *sync.Once
 	initEpoch int
 	initErr   error
-	subnets   []*subnet
+	*networkConfiguration
```

Comment:
Any change to data-structures that is saved in kv-store must be handled with care. Changing `subnets` to `networkConfiguration` will result in existing networks in the store to become invalid and will result in json unmarshalling issues when the docker daemon is restarted.
Adding a new variable is fine. Hence, I would suggest you to retain the `subnets` in its place and just add a boolean `disableDefaultGW`.

Even better, it would be good to avoid saving `disableDefaultGW`, instead save the options that we get in CreateNetwork as map[string]string. So that we need not add individual options if we have to add them in the future.

Can you please take care of that ?